libs/core/src/lib/carousel/carousel.component.ts

```diff
@@ -237,18 +241,18 @@ export class CarouselComponent implements OnInit, AfterContentInit, AfterViewIni
         } else {
             this.leftButtonDisabled = true;
             this.rightButtonDisabled = true;
-        }
-        this._carouselService.activeChange.subscribe((event) => this._onSlideSwipe(event));
-        this._carouselService.dragStateChange.subscribe((event) => this._onSlideDrag(event));
-
-        this._slidesCopy = this.slides.toArray().slice();
 
-        // Disable swipe when there is no carousel item
-        if (this.slides.length === 0) {
+            // Disable swipe when there is no carousel item
             this.swipeEnabled = false;
             this.navigation = false;
         }
 
+        this._slidesCopy = this.slides.toArray().slice();
```

Comment:
Why `.slice()`?

Reply:
placed comment now.
In looped carousel, original array gets changes with shift and unshift. for carousel indicator, have to keep a copy of original array.

Reply:
removed un-necessary slice of array.